lec05-dplyr.Rmd

```diff
@@ -1,4 +1,510 @@
+them with your collaborators or for archival.
+
+Similar to the `read_csv()` function used for reading CSV files into R, there is
+a `write_csv()` function that generates CSV files from data frames.
+
+Before using `write_csv()`, we are going to create a new folder, `data`,
+in our working directory that will store this generated dataset. We don't want
+to write generated datasets in the same directory as our raw data. It's good
+practice to keep them separate. The `data` folder should only contain the raw,
+unaltered data, and should be left alone to make sure we don't delete or modify
+it. In contrast, our R code will generate the contents of the `data`
+directory, so even if the files it contains are deleted, we can always
+re-generate them.
+
+Use the `getwd()` function to find out which is the current working directory.
+
+```{r}
+getwd()
+```
+
+Navigate to this directory in your file browser and create a folder called data output.
+
+Alternatively, you could use R to create this directory.
+
+```{r}
+dir.create("data")
+
+# To suppress the warning, we could do
+dir.create("data", showWarnings = FALSE)
+
+# Another alternative would be to use a conditional expression, which only
+# creates the directory *if* it does not already exist. The syntax here is
+# similar to the for loop we created in the second lecture.
+if (!dir.exists('data')) {
+    dir.create("data")
+}
+```
+
+We are going to prepare a cleaned up version of the data without NAs. Let's 
+start by removing observations for which the `species_id` is missing. Let's also
+remove observations for which `weight` and the `hindfoot_length` are missing.
+This dataset should also only contain observations of animals for which the sex
+has been determined:
+
+```{r}
+surveys_complete <- surveys %>%
+  filter(!is.na(species_id),       # remove missing species_id
+         !is.na(weight),           # remove missing weight
+         !is.na(hindfoot_length),  # remove missing hindfoot_length
+         !is.na(sex))              # remove missing sex
+
+# This expression is a succinct alternative to the above
+surveys_complete_comcas <- surveys %>%
+    filter(complete.cases(species_id, weight, hindfoot_length, sex))
+
+# This is even briefer, but omits observations with NA in *any* column.
+# There is no way to control which columns to use, but it is common to want
+# to exclude all NAs, which in our case corresponds to the columns listed above.
+surveys_complete_naomit <- na.omit(surveys)
+
+# Compare the dimensions of the original and the cleaned data frame
+dim(surveys)
+dim(surveys_complete)
+dim(surveys_complete_comcas)
+dim(surveys_complete_naomit)
+```
+
+Now that our dataset is ready, we can save it as a CSV file in our `data`
+folder.
+
+```{r}
+write_csv(surveys_complete, path = "data/surveys_complete.csv")
+```
+
+
+## Team Challenges
+
+These are four excersices in exploratory data analyses, where you will train to 
```

Comment:
`analyses, you will learn` or `analyses, which will train you to`

Reply:
s/exersices/exercises/